### PR TITLE
feat(visual mode): put visual text into `vim.ui.input`

### PR DESCRIPTION
**Pull request by phanen**
support visual mode, here is a minimal config (use `lazy.nvim` as package manager):

```lua
return {
    "lalitmee/browse.nvim",
    dependencies = { "nvim-telescope/telescope.nvim" },
    keys = {
      {
        mode = { "n", "x" },
        "<leader>S",
        function() require("browse").open_bookmarks() end,
      },
      {
        mode = { "n", "x" },
        "<leader>B",
        function() require("browse").browse() end,
      } },

    opts = {
      bookmarks = {
        ["github_code_search"] = "https://github.com/search?q=%s&type=code",
        ["github_repo_search"] = "https://github.com/search?q=%s&type=repositories",
      },
    },
},
```

### CODE REVIEW

File: lua/browse/input.lua [hunk 1]
```diff
@@ -18,6 +18,6 @@ M.search_input = utils.callback_search(function(input)
     end
 
     return path
-end)
+end, {})
```
Comment (lalitmee) [inline lua/browse/input.lua:21]: Can we also support this here? It is when someone calls `input_search` directly.

File: lua/browse/utils.lua [hunk 2]
```diff
@@ -67,7 +67,9 @@ end
 -- a generic searching function used everywhere
 M.search = function(target_fn, opts)
     local prompt = opts and opts.prompt or "Search String:"
-    vim.ui.input({ prompt = prompt, kind = "browse" }, function(input)
+    local default = opts and opts.visual_text or ""
+    print(default)
```
Comment (lalitmee) [inline lua/browse/utils.lua]: remove this

File: lua/browse/utils.lua [hunk 3]
```diff
@@ -79,18 +81,29 @@ end
...
-    return function()
+    return function(visual_text)
+        opts.visual_text = visual_text
         M.search(function(input)
             return string.format(format, input)
         end, opts)
     end
 end
 
+M.get_visual_text = function()
```
Comment (lalitmee) [inline lua/browse/utils.lua:100]: write some documentation in the form of comments for the function about what is being done in the process.

File: lua/browse/bookmarks.lua [hunk 4]
```diff
@@ -7,13 +7,16 @@ local action_state = require("telescope.actions.state")
...
 local defaults = require("browse.config")
+local get_visual_text = require("browse.utils").get_visual_text
 
 local M = {}
 
 -- search bookmarks
 M.search_bookmarks = function(config)
     config = config or {}
     local bookmarks = config["bookmarks"] or defaults.opts["bookmarks"] or {}
+    -- skip if get it before
```
Comment (lalitmee) [inline lua/browse/bookmarks.lua]: no need of this comment here, its obvious

**Review by lalitmee — CHANGES_REQUESTED** [review]

File: lua/browse/input.lua (hunk 1 above)
Comment (phanen) [reply lua/browse/input.lua:21]: According to <https://github.com/lalitmee/browse.nvim/pull/16/commits/abc8d0948b9e5ec047fa43dff743754b001dae0b>, now `input_search` can be called by user with visual mode support.
e.g.
```lua
    keys = {
      {
        mode = { "n", "x" },
        "<leader>I",
        function() require("browse").input_search() end,
      },
    },
```

**Review by lalitmee — APPROVED** [review]

**Comment by lalitmee** [comment]
@phanen, thanks.

**Pull request merged**